sdk/data/azcosmos/emulator_cosmos_item_test.go

```diff
@@ -503,6 +503,211 @@ func TestItemIdEncodingComputeGW(t *testing.T) {
 	verifyEncodingScenario(t, container, "ComputeGW-IdWithUnicodeCharacters", "WithUnicode鱀", http.StatusCreated, http.StatusOK, http.StatusOK, http.StatusNoContent)
 }
 
+func TestItemCRUDHierarchicalPartitionKey(t *testing.T) {
+	emulatorTests := newEmulatorTests(t)
+	client := emulatorTests.getClient(t, newSpanValidator(t, spanMatcher{
+		ExpectedSpans: []string{},
+	}))
+
+	database := emulatorTests.createDatabase(t, context.TODO(), client, "itemCRUDHierarchicalPartitionKey")
+	defer emulatorTests.deleteDatabase(t, context.TODO(), database)
+	properties := ContainerProperties{
+		ID: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/id", "/type"},
+		},
```

Comment:
Ah, I think this is where the issue is arising. When creating a multi-partition-key container, you need to set the `kind` field to `MultiHash`. We don't currently support that in Go, so you'll need to add support for that. For example, you can see the [JS SDK has a `kind` property](https://github.com/Azure/azure-sdk-for-js/blob/497321f0400ba4a2adfbd8f59db62d7574b52824/sdk/cosmosdb/cosmos/src/documents/PartitionKeyDefinition.ts#L20-L22). We'd need to add this field in a way that doesn't break existing users. It might be worth a custom `MarshalJSON`/`UnmarshalJSON` implementation here, which automatically sets `kind` based on the number of partition key paths 🤔

Reply:
Done! I've added custom (de)serialization where the value for `Kind` is inferred if its value is an empty string.

Reply:
The tests are still failing, but thankfully in new places - progress!
The current issue seems to be that the type of the PartitionKey in the container doesn't match that which is supplied in the headers, I guess that `PartitionKey` also needs to be extended with `Kind`.

Reply:
I think those are some typos in your test code actually, mismatching some places where you're specifying the "alpha" partition key but sending the "alpha" document as content.